### PR TITLE
Add escaping support for System.Text.Formatting

### LINKED ISSUE #61: Composite format should support escaping "{"
The following test will fail:

            var formatter = new StringFormatter();
            formatter.Format("{{{0}", 1);
            Assert.Equal("{1", formatter.ToString());

### PR DESCRIPTION
**Pull request by alexanderkozlenko**
Resolves #61

### CODE REVIEW

File: src/System.Text.Formatting/tests/CompositeFormattingTests.cs [hunk 1]
```diff
@@ -37,5 +37,34 @@ public void CompositeFormattingFormatStrings()
...
+
+        [Fact]
+        public void CompositeFormattingEscaping()
+        {
+            var format = "}}a {0} b {0} c {{{0}}} d {{e}} {{";
+            var pool = new ManagedBufferPool<byte>(1024);
+            var formatter = new StringFormatter(pool);
+            formatter.Format(format, 1);
+
+            Assert.Equal(string.Format(format, 1), formatter.ToString());
```
Comment (KrzysztofCwalina) [inline src/System.Text.Formatting/tests/CompositeFormattingTests.cs]: Should it explicitly use the invariant culture?

Comment (alexanderkozlenko) [inline src/System.Text.Formatting/tests/CompositeFormattingTests.cs]: It seems there is no way to specify culture for StringFormatter, thus StringFormatter and string.Format will use a derivative of invariant culture, which is defined for current thread in test class constructor.

Comment (KrzysztofCwalina) [inline src/System.Text.Formatting/tests/CompositeFormattingTests.cs]: string.Format uses the "current culture", not invariant culture. I think the test should use: string.Format(CultureInfo.InvariantCulture, format, 1)

File: src/System.Text.Formatting/tests/CompositeFormattingTests.cs [hunk 2]
```diff
@@ -37,5 +37,34 @@ public void CompositeFormattingFormatStrings()
...
+            var format = "}}a {0} b {0} c {{{0}}} d {{e}} {{";
+            var pool = new ManagedBufferPool<byte>(1024);
+            var formatter = new StringFormatter(pool);
+            formatter.Format(format, 1);
+
+            Assert.Equal(string.Format(CultureInfo.InvariantCulture, format, 1), formatter.ToString());
+        }
+
+        [Fact]
+        public void CompositeFormattingEscapingMissingStartBracket()
```
Comment (KrzysztofCwalina) [inline src/System.Text.Formatting/tests/CompositeFormattingTests.cs]: Should this test be called ...MissingEndBracket?

File: src/System.Text.Formatting/tests/CompositeFormattingTests.cs [hunk 3]
```diff
@@ -37,5 +37,34 @@ public void CompositeFormattingFormatStrings()
...
+        {
+            var pool = new ManagedBufferPool<byte>(1024);
+            var formatter = new StringFormatter(pool);
+
+            Assert.Throws<Exception>(() => formatter.Format("{{0}", 1));
+        }
+
+        [Fact]
+        public void CompositeFormattingEscapingMissingEndBracket()
```
Comment (KrzysztofCwalina) [inline src/System.Text.Formatting/tests/CompositeFormattingTests.cs]: Should this test be called ...MissingStartBracket?

**Comment by KrzysztofCwalina** [comment]
looks great! Thanks.

**Pull request merged**